lms/static/scripts/frontend_apps/components/AutoGradingConfigurator.tsx

```diff
@@ -0,0 +1,215 @@
+
+export type AutoGradingConfig = {
+  /** Whether auto grading is active for the assignment or not */
+  active?: boolean;
+
+  /**
+   * - all_or_nothing: students need to meet a minimum value, making them get
+   *                   either 0% or 100%
+   * - scaled: students may get a proportional grade based on the amount of
+   *           annotations. If requirement is 4, and they created 3, they'll
+   *           get a 75%
+   */
+  gradingType: GradingType;
+
+  /**
+   * - cumulative: both annotations and replies will be counted together for
+   *               the grade calculation
+   * - separately: students will have different annotation and reply goals.
+   */
+  activityCalculation: 'cumulative' | 'separately';
+
+  /**
+   * Required number of annotations if activityCalculation is 'separately' or
+   * annotations+replies otherwise
+   */
+  requiredAnnotations: number;
+
+  /**
+   * Required number of replies if activityCalculation is 'separately'
+   */
+  requiredReplies?: number;
+};
+
+type AnnotationsGoalInputGroupProps = {
+  children?: ComponentChildren;
+  gradingType: GradingType;
+  value: number;
+  onChange: (newValue: number) => void;
+
+  /** Minimum required value for the input. Defaults to 1 */
+  min?: number;
+};
+
+/**
+ * Controls containing a number input to set the amount of required annotations
+ * or replies
+ */
+function AnnotationsGoalInputGroup({
+  children,
+  gradingType,
+  value,
+  onChange,
+  min = 1,
+}: AnnotationsGoalInputGroupProps) {
+  const inputId = useId();
+
+  return (
+    <div className="flex gap-2 items-center">
+      <label
+        className="grow flex justify-between items-center"
+        htmlFor={inputId}
+      >
+        {children}
+        <span className="uppercase font-semibold">
+          {gradingType === 'all_or_nothing' ? 'Minimum' : 'Goal'}
+        </span>
+      </label>
+      <Input
+        id={inputId}
+        classes="max-w-14"
+        type="number"
+        required
+        min={min}
+        step={1}
+        value={value}
+        onChange={e => onChange(Number((e.target as HTMLInputElement).value))}
+      />
+    </div>
+  );
+}
+
+export type AutoGradingConfiguratorProps = {
+  config: AutoGradingConfig;
+  updateAutoGradingConfig: (newConfig: AutoGradingConfig) => void;
+};
+
+/**
+ * Allows instructors to enable auto grading for an assignment, and provide the
+ * configuration to determine how to calculate every student's grade.
+ */
+export default function AutoGradingConfigurator({
+  config,
+  updateAutoGradingConfig,
+}: AutoGradingConfiguratorProps) {
+  const {
+    active = false,
+    gradingType,
+    activityCalculation,
+    requiredAnnotations,
+    requiredReplies = 0,
+  } = config;
+  const updatePartialConfig = useCallback(
+    (partialConfig: Partial<AutoGradingConfig>) =>
+      updateAutoGradingConfig({ ...config, ...partialConfig }),
+    [config, updateAutoGradingConfig],
+  );
```

Comment:
```suggestion
  const updateConfig = useCallback(
    (newConfig: Partial<AutoGradingConfig>) =>
      updateAutoGradingConfig({ ...config, ...newConfig }),
    [config, updateAutoGradingConfig],
  );
```

This function always sets the full config, but the argument is a partial config that contains new configuration to merge with the existing value.